textdb/textdb-web/src/main/java/edu/uci/ics/textdb/web/request/beans/RegexSourceBean.java

```diff
@@ -65,18 +65,19 @@ public void setDataSource(String dataSource) {
     public boolean equals(Object other) {
         if (other == null) return false;
         if (other == this) return true;
-        if (!(other instanceof OperatorBean)) return false;
+        if (!(other instanceof RegexSourceBean)) return false;
         RegexSourceBean regexSourceBean = (RegexSourceBean) other;
         return new EqualsBuilder()
+                .appendSuper(super.equals(regexSourceBean))
                 .append(regex, regexSourceBean.getRegex())
                 .append(dataSource, regexSourceBean.getDataSource())
-                .isEquals() &&
-                super.equals(regexSourceBean);
+                .isEquals();
     }
 
     @Override
     public int hashCode() {
         return new HashCodeBuilder(17, 31)
+                .append(super.hashCode())
```

Comment:
Hi Kishore and Flavio, here's my opinion: I think we all agree that this change won't hurt the functionality. It's just a matter of choice. 

So the way `equals()` works is it checks the hashcodes first, if the hashcodes are different, `false` will be returned immediately and the lines in `equals()` won't be executed at all. 

So although Kishore is right about superclass's equals will be called, and eventually producing the same result, having distinct hashcodes for different object is generally good, and will improve performance (maybe just slightly a little bit). But either way the functionality is the same.